vortex-gpu/src/rle_decompress.rs

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+use std::sync::Arc;
+
+use cudarc::driver::{CudaContext, DeviceRepr, LaunchConfig, PushKernelArg, ValidAsZeroBits};
+use vortex_array::arrays::PrimitiveArray;
+use vortex_array::validity::Validity;
+use vortex_array::{ArrayRef, IntoArray, ToCanonical};
+use vortex_buffer::BufferMut;
+use vortex_dtype::{NativePType, UnsignedPType, match_each_native_ptype};
+use vortex_error::{VortexExpect, VortexResult, vortex_err, vortex_panic};
+use vortex_fastlanes::RLEArray;
+
+use crate::take::cuda_take_kernel;
+
+pub fn cuda_rle_decompress(
+    rle: &RLEArray,
+    ctx: Arc<CudaContext>,
+) -> VortexResult<Option<ArrayRef>> {
+    match_each_native_ptype!(rle.values().dtype().as_ptype(), |V| {
+        // RLE indices are always u16 (or u8 if downcasted).
+        match rle.indices().dtype().as_ptype() {
+            PType::U8 => cuda_rle_decompress_typed(
+                rle.indices().to_primitive().as_slice::<u8>(),
+                rle.values().to_primitive().as_slice::<V>(),
+                ctx,
+            )
+            .map(Some),
+            PType::U16 => cuda_rle_decompress_typed(
+                rle.indices().to_primitive().as_slice::<u16>(),
+                rle.values().to_primitive().as_slice::<V>(),
+                ctx,
+            )
+            .map(Some),
```

Comment:
Why wrap the `ArrayRef` in an `Option`?

Reply:
Didn't we decide to make decompression infallible? Should this return a result or panic.